src/domain/consents/{ID}.ts

```diff
@@ -0,0 +1,88 @@
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Ahan Gupta <ahangupta.96@gmail.com>
+
+ --------------
+ ******/
+import { Consent } from '../../model/consent'
+import { Scope } from '../../model/scope'
+import { consentDB, scopeDB } from '../../lib/db'
+import { IncorrectChallengeError, IncorrectStatusError } from '../errors'
+import { PutConsentsRequest } from '@mojaloop/sdk-standard-components'
+import { thirdPartyRequest } from '../../lib/requests'
+import { Enum } from '@mojaloop/central-services-shared'
+import { Request } from '@hapi/hapi'
+import { ExternalScope, convertScopesToExternal } from '../../lib/scopes'
+
+export interface ConsentCredential {
+  credentialId: string;
+  credentialStatus: 'ACTIVE';
+  credentialPayload: string;
+}
+
+export async function retrieveValidConsent (consentId: string, requestChallenge: string): Promise<Consent> {
+  const consent: Consent = await consentDB.retrieve(consentId)
+  if (consent.credentialChallenge !== requestChallenge) {
+    throw new IncorrectChallengeError(consentId)
+  }
+  return consent
+}
+
+export async function checkCredentialStatus (credentialStatus: string, consentId: string): Promise<void> {
+  if (credentialStatus !== 'PENDING') {
+    throw new IncorrectStatusError(consentId)
+  }
+}
+
+/*
+ * Updates the consent resource in the database with incoming request's
+ * credential attributes.
+ */
+export async function updateConsentCredential (consent: Consent, credential: ConsentCredential): Promise<number> {
+  consent.credentialId = credential.credentialId
+  consent.credentialStatus = credential.credentialStatus
+  consent.credentialPayload = credential.credentialPayload
+  return consentDB.update(consent)
+}
+
+export async function putConsents (consent: Consent, signature: string, publicKey: string, request: Request): Promise<void> {
+  /* Retrieve the scopes pertinent to this consentId and populate the scopes accordingly. */
+  const scopes: Scope[] = await scopeDB.retrieveAll(consent.id)
+  const externalScopes: ExternalScope[] = convertScopesToExternal(scopes)
+  const consentBody: PutConsentsRequest = {
```

Comment:
Abstract 68-85 into a separate function and return the consentBody.


Use the spread operator to copy the properties from the consent over to the consentBody and overload as and when necessary.